.github/workflows/deploy.yaml

```diff
@@ -494,6 +494,21 @@ jobs:
           slack-message-id: ${{ needs.init.outputs.slack-message-id }}
       - name: Checkout
         uses: actions/checkout@v4
+      - name: Ensure commit belongs to release branch
+        if: ${{ inputs.stage == 'production' }}
+        run: |
+          # Fetch release branch candidates (! to allow non-existence)
+          ! git fetch origin master
+          ! git fetch origin main
+
+          # Check which branches contain the current HEAD (! to allow no grep matches)
+          ! IS_MAIN=$(git branch -r --contains HEAD --format '%(refname)' | grep -E '^refs/remotes/origin/(master|main)$')
```

Comment:
pedantic: with this, you can release with a very old commit as long as the commit exists on master/main

Reply:
Yeah it is not perfect. There is also typically just one of main/master that has branch protection so it's not stopping anyone from force pushing to the unprotected one and releasing whatever they want 😁

Reply:
Haha, yeah, didn't think about that!

I don't know the extent of this actions use, but overall, I'd say it looks good to me 👍 - not perfect, but doesn't hurt

Reply:
Let's say it's more a safeguard against accidentally pushing a tag rather than preventing malicious behavior. If a disgruntled employee wants to get something into prod without review, I would assume there is 100 other ways to do so, so we cannot prevent it 100%